tools/dotnet-linker/Steps/GenerateMainStep.cs

```diff
@@ -31,16 +31,13 @@ protected override void TryEndProcess ()
 
 				contents.AppendLine ("#include <stdlib.h>");
 				contents.AppendLine ();
-				contents.AppendLine ("extern \"C\" const char *xamarin_icu_dat_file_name;");
-				contents.AppendLine ();
 				contents.AppendLine ("static void xamarin_initialize_dotnet ()");
 				contents.AppendLine ("{");
 				if (Configuration.InvariantGlobalization) {
 					contents.AppendLine ("\tsetenv (\"DOTNET_SYSTEM_GLOBALIZATION_INVARIANT\", \"1\", 1);");
```

Comment:
for a future PR: this could be a runtime config option too: `System.Globalization.Invariant` (https://learn.microsoft.com/en-us/dotnet/core/runtime-config/globalization)

or actually we might not even need to do anything if the `InvariantGlobalization` msbuild property is set it should get turned into the runtime config automatically?

Reply:
Yes, this is from the early .NET 5 days, so the runtime config option probably didn't work then.

Reply:
The `DOTNET_SYSTEM_GLOBALIZATION_HYBRID` environment variable is pretty new though:

https://github.com/xamarin/xamarin-macios/commit/3f6d43c75649a85e47fd23d6d48c8d5a87257d7b

so maybe @mkhamoyan knows if these environment variables are really needed or not?

Reply:
I added `DOTNET_SYSTEM_GLOBALIZATION_HYBRID` following to the logic done for Invariant mode. Not sure if these env variables are really needed or not.

Reply:
The runtime seems to first look into the runtime config switches and then as a fallback into env variables:
https://github.com/dotnet/runtime/blob/1487d46e4ff7731b52af92b1228373a2a66ed6bf/src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs#L15-L17
which calls into: https://github.com/dotnet/runtime/blob/1487d46e4ff7731b52af92b1228373a2a66ed6bf/src/libraries/System.Private.CoreLib/src/System/AppContextConfigHelper.cs#L13-L22

Do you want me to remove setting up env variables explicitly in `xamarin_initialize_dotnet`?

Reply:
> Do you want me to remove setting up env variables explicitly in xamarin_initialize_dotnet?

we can do that in a separate PR after verifying that the switches to end up in runtimeconfig.bin etc. but yeah, it'd be nice to reduce custom logic

Reply:
I checked this by building `MySingleView` and inspecting `MySingleView.runtimeconfig.json`:
```json
    "configProperties": {
      "Microsoft.Extensions.DependencyInjection.VerifyOpenGenericServiceTrimmability": true,
      "System.AggressiveAttributeTrimming": false,
      "System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization": false,
      "System.Diagnostics.Debugger.IsSupported": true,
      "System.Diagnostics.Tracing.EventSource.IsSupported": false,
      "System.Globalization.Invariant": false,
      "System.Globalization.Hybrid": false,
      "System.Net.Http.EnableActivityPropagation": false,
      "System.Net.Http.UseNativeHttpHandler": true,
      "System.Reflection.NullabilityInfoContext.IsSupported": false,
      "System.Resources.ResourceManager.AllowCustomResourceTypes": false,
      "System.Resources.UseSystemResourceKeys": false,
      "System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported": false,
      "System.Runtime.InteropServices.BuiltInComInterop.IsSupported": false,
      "System.Runtime.InteropServices.EnableConsumingManagedCodeFromNativeHosting": false,
      "System.Runtime.InteropServices.EnableCppCLIHostActivation": false,
      "System.Runtime.InteropServices.Marshalling.EnableGeneratedComInterfaceComImportInterop": false,
      "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization": false,
      "System.StartupHookProvider.IsSupported": false,
      "System.Text.Encoding.EnableUnsafeUTF7Encoding": false,
      "System.Text.Json.JsonSerializer.IsReflectionEnabledByDefault": true,
      "System.Threading.Thread.EnableAutoreleasePool": true,
      "System.Linq.Expressions.CanEmitObjectArrayDelegate": false,
      "ICU_DAT_FILE_PATH": "icudt.dat"
    }
```

Reply:
I have included the clean up in this PR, will see what the tests say